Src/WitsmlExplorer.Api/Workers/Copy/CopyLogDataWorker.cs

```diff
@@ -275,33 +276,50 @@ private static WitsmlData CollateData(List<string[]> oldRows, double index)
 
         private async Task VerifyTargetHasRequiredLogCurveInfos(WitsmlLog sourceLog, IEnumerable<string> sourceMnemonics, WitsmlLog targetLog)
         {
-            List<WitsmlLogCurveInfo> newLogCurveInfos = new();
-            foreach (string mnemonic in sourceMnemonics.Where(mnemonic =>
-            !string.Equals(targetLog.IndexCurve.Value, mnemonic, StringComparison.OrdinalIgnoreCase) &&
-            !string.Equals(sourceLog.IndexCurve.Value, mnemonic, StringComparison.OrdinalIgnoreCase)))
+            List<WitsmlLogCurveInfo> logCurveInfosToUpdate = new();
+            foreach (string mnemonic in sourceMnemonics)
             {
-                if (targetLog.LogCurveInfo.All(lci => !string.Equals(lci.Mnemonic, mnemonic, StringComparison.OrdinalIgnoreCase)))
+                bool isIndexCurve = string.Equals(mnemonic, sourceLog.IndexCurve.Value, StringComparison.OrdinalIgnoreCase);
```

Comment:
sourceLog.IndexCurve.Value
targetLog.IndexCurve.Value
Is that assured that IndexCurve is not null in both cases?

Reply:
`indexCurve` is a required field, so it should not be null.
This is from the xsd:
`<xsd:element name="indexCurve" type="witsml:str32" minOccurs="1" maxOccurs="1">`